src/wallet/walletdb.h

```diff
@@ -269,6 +270,60 @@ class CWalletDB : public CDB
     //! write the hdchain model (external chain child index counter)
     bool WriteHDChain(const CHDChain& chain);
 
+#ifdef ENABLE_EXODUS
+    template<typename K, typename V>
+    bool WriteExodusEntry(const K& k, const V& v) {
+        return Write(std::make_pair(std::string("exodus_entry"), k), v);
+    }
+
+    template<typename K, typename V>
+    bool ReadExodusEntry(const K& k, V& v) {
+        return Read(std::make_pair(std::string("exodus_entry"), k), v);
+    }
+
+    template<typename K>
+    bool HasExodusEntry(const K& k) {
+        return Exists(std::make_pair(std::string("exodus_entry"), k));
+    }
+
+    template<typename K, typename V>
+    void ListExodusEntries(std::list<V>& entries) {
```

Comment:
This way it will discard key. My idea is make this function to accept a callback function instead. The benefits with this way is we can also pass key to a callback too. Also when we want to apply filter it can do in one loop.

Reply:
Good idea